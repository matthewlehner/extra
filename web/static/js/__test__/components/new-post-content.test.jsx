import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter, Route } from "react-router-dom";

import NewPostContent from "components/new-post-content";

describe("NewPostContent component", () => {
  it("renders properly", () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={["/collections/1/new"]}>
        <Route
          path="/collections/:id/new"
          render={props => (
            <NewPostContent {...props} />
          )}
        />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
