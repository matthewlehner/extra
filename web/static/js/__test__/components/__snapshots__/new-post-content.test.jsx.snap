// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPostContent component renders properly 1`] = `
<div
  className="modal__overlay"
  onClick={[Function]}
>
  <div
    className="modal__container"
  >
    <header
      className="modal__header"
    >
      <a
        className="modal__dismiss"
        href="/collections/1"
        onClick={[Function]}
      >
        Ã—
      </a>
      <h2>
        Create new post
      </h2>
    </header>
    <div
      className="modal__body"
    >
      <form>
        <div
          className="form__control-group"
        >
          <label
            className="form__control-label"
            htmlFor="post_content_post_collection_id"
          >
            Collection
          </label>
          <div
            className="select"
          >
            <select
              className="form__control"
              id="post_content_post_collection_id"
              name="post_content[post_collection_id]"
              required="required"
            >
              <option
                value="1"
              >
                Inspirational Posts
              </option>
            </select>
          </div>
        </div>
        <div
          className="form__control-group"
        >
          <label
            className="form__control-label"
            htmlFor="post_content_body"
          >
            Content
          </label>
          <textarea
            className="form__control"
            id="post_content_body"
            name="post_content[body]"
            required="required"
          />
        </div>
        <div
          className="form__control-group"
        >
          <span
            className="form__control-label"
          >
            Channel
          </span>
          <div
            className="form__collection-wrapper"
          >
            <label
              htmlFor="post_content_templates_0_active"
            >
              <input
                className="form__control"
                id="post_content_templates_0_active"
                name="post_content[templates][0][active]"
                type="checkbox"
                value="true"
              />
              <span
                className="form__indicator form__indicator_checkbox"
              >
                <svg
                  className="icon icon_checkmark"
                >
                  <use
                    xlinkHref="/images/app.svg#checkmark-icon"
                  />
                </svg>
              </span>
              matthewpearse
            </label>
          </div>
        </div>
        <div
          className="form__actions"
        >
          <a
            className="button button_cancel"
            href="/collections/1"
            onClick={[Function]}
          >
            Cancel
          </a>
          <button
            className="button"
            type="submit"
          >
            Create Post
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
